pkg/pgreplay/parse.go

```diff
@@ -76,6 +77,47 @@ func ParseJSON(jsonlog io.Reader) (items chan Item, errs chan error, done chan e
 	return
 }
 
+func ParseCsvLog(csvlog io.Reader) (items chan Item, errs chan error, done chan error) {
+	reader := csv.NewReader(csvlog)
+	unbounds := map[SessionID]*Execute{}
+	parsebuffer := make([]byte, MaxLogLineSize)
+	items, errs, done = make(chan Item, ItemBufferSize), make(chan error), make(chan error)
+
+	go func() {
+		for {
+			logline, err := reader.Read()
+			if err == io.EOF {
+				break
+			}
+			if err != nil {
+				logLinesErrorTotal.Inc()
+				errs <- err
+			}
+			item, err := ParseCsvItem(logline, unbounds, parsebuffer)
+			if err != nil {
+				logLinesErrorTotal.Inc()
+				errs <- err
+			}
+
+			if item != nil {
+				logLinesParsedTotal.Inc()
+				items <- item
+			}
+		}
+
+		// Flush the item channel by pushing nil values up-to capacity
+		for i := 0; i < ItemBufferSize; i++ {
+			items <- nil
+		}
+
```

Comment:
Looking at the implementation of ParseJSON I don't _think_ this is necessary?

Reply:
Hi @chill, I agree with you 👍🏻, since we can also read from a closed channel we can delete this part of flushing the buffer. On the `database.go` & `streamer.go` I'm going to keep the null validations in case a null value is being leaked somewhere in the code.

I'll delete the for loop block on `ParseCsvLog` & `ParseErrlog` 🫡